kerastuner/tuners/ultraband.py

```diff
@@ -12,28 +12,283 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"Variation of HyperBand algorithm."
 
-from ...engine import tuner as tuner_module
-from ...engine import oracle as oracle_module
+import queue
+import random
+from ..engine import tuner as tuner_module
+from ..engine import oracle as oracle_module
 
 
 class UltraBandOracle(oracle_module.Oracle):
-    # TODO
-    pass
+    """Oracle class for UltraBand.
+
+        Args:
+        max_trials: Int. Total number of trials
+            (model configurations) to test at most.
+            Note that the oracle may interrupt the search
+            before `max_trial` models have been tested.
+        seed: Int. The random seed. If None, it would use a random number.
+        factor: Int. The factor of the change of number of epochs
+            and the number of models per bracket.
+        min_epochs: Int. The minimum number of epochs to train a model.
+        max_epochs: Int. The maximum number of epochs to train a model.
+    """
+
+    def __init__(self,
+                 max_trials=200,
+                 seed=None,
+                 factor=3,
+                 min_epochs=3,
+                 max_epochs=10):
+        super().__init__()
+        if min_epochs >= max_epochs:
+            raise ValueError('max_epochs needs to be larger than min_epochs.')
+        if factor < 2:
+            raise ValueError('factor needs to be a int larger than 1.')
+        self.trials = max_trials
```

Comment:
Pass max_trials to `super().__init__()` instead

Reply:
The max_trial is removed from the oracle.